Autoprefixer setting -ms-grid row and column to opposite actual values

https://stackoverflow.com/questions/52425765/css-grid-ms-grid-row-and-column-being-set-opposite-to-actual-value

I posted a question regarding this here. With a little more investigation I think this may be a autoprefixer issue. As when I manually add the -ms- syntax (I am compiling from Sass) into the compiled CSS file. It works correctly.

Does this need remedying?

```
+-- autoprefixer@9.1.5
| +-- browserslist@4.1.1
| | +-- electron-to-chromium@1.3.70
| | `-- node-releases@1.0.0-alpha.11
| |   `-- semver@5.5.1
| +-- caniuse-lite@1.0.30000885
| `-- postcss@7.0.2
|   +-- chalk@2.4.1
|   | `-- ansi-styles@3.2.1
|   +-- source-map@0.6.1
|   `-- supports-color@5.5.0
`-- gulp-autoprefixer@5.0.0
  `-- autoprefixer@8.6.5
    `-- browserslist@3.2.8

PS C:\Users\escull\Documents\GitHub\Grid-Layout> npm ls | grep autoprefixer
npm ERR! extraneous: autoprefixer@9.1.5 C:\Users\escull\Documents\GitHub\Grid-Layout\node_modules\autoprefixer
+-- autoprefixer@9.1.5 extraneous
+-- gulp-autoprefixer@5.0.0
| +-- autoprefixer@8.6.5
```

--- a/lib/hacks/grid-area.js
+++ b/lib/hacks/grid-area.js
@@ -25,6 +25,8 @@ class GridArea extends Declaration {
     })
 
     utils.warnTemplateSelectorNotFound(decl, result)
+    utils.warnIfGridRowColumnExists(decl, result)
+
     return undefined
   }
 }

--- a/lib/hacks/grid-utils.js
+++ b/lib/hacks/grid-utils.js
@@ -720,6 +720,30 @@ function warnTemplateSelectorNotFound (decl, result) {
   }
 }
 
+/**
+ * warn user if both grid-area and grid-(row|column)
+ * declarations are present in the same rule
+ * @param  {Declaration} decl
+ * @param  {Result} result
+ * @return {void}
+ */
+function warnIfGridRowColumnExists (decl, result) {
+  let rule = decl.parent
+  let decls = []
+  rule.walkDecls(/^grid-(row|column)/, d => {
+    decls.push(d)
+  })
+  if (decls.length > 0) {
+    decls.forEach(d => {
+      d.warn(result,
+        `You already have grid-area declaration present in the rule. ` +
+        `You should use either grid-area or ${ d.prop }, not both`)
+    })
+  }
+
+  return undefined
+}
+
 // Gap utils
 
 function getGridGap (decl) {
@@ -852,5 +876,6 @@ module.exports = {
   getGridGap,
   warnGridGap,
   warnTemplateSelectorNotFound,
+  warnIfGridRowColumnExists,
   inheritGridGap
 }

--- a/lib/hacks/grid-utils.js
+++ b/lib/hacks/grid-utils.js
@@ -736,7 +736,7 @@ function warnIfGridRowColumnExists (decl, result) {
   if (decls.length > 0) {
     decls.forEach(d => {
       d.warn(result,
-        `You already have grid-area declaration present in the rule. ` +
+        `You already have a grid-area declaration present in the rule. ` +
         `You should use either grid-area or ${ d.prop }, not both`)
     })
   }

--- a/test/autoprefixer.test.js
+++ b/test/autoprefixer.test.js
@@ -565,6 +565,22 @@ describe('hacks', () => {
     expect(result.warnings()).toEqual([])
   })
 
+  it('warns if rule has both grid-area and grid-(row|column) decls', () => {
+    let input = read('grid-area')
+    let instance = prefixer('grid-area')
+    let result = postcss([instance]).process(input)
+    expect(result.warnings()
+      .map(i => i.toString()))
+      .toEqual([
+        'autoprefixer: <css input>:28:3: You already have grid-area ' +
+        'declaration present in the rule. You should use either ' +
+        'grid-area or grid-row, not both',
+        'autoprefixer: <css input>:29:3: You already have grid-area ' +
+        'declaration present in the rule. You should use either ' +
+        'grid-area or grid-column, not both'
+      ])
+  })
+
   it('warns if rule with grid-area has no parent with grid-template', () => {
     let input = read('grid-template-areas')
     let instance = prefixer('grid-area')

--- a/test/autoprefixer.test.js
+++ b/test/autoprefixer.test.js
@@ -572,10 +572,10 @@ describe('hacks', () => {
     expect(result.warnings()
       .map(i => i.toString()))
       .toEqual([
-        'autoprefixer: <css input>:28:3: You already have grid-area ' +
+        'autoprefixer: <css input>:28:3: You already have a grid-area ' +
         'declaration present in the rule. You should use either ' +
         'grid-area or grid-row, not both',
-        'autoprefixer: <css input>:29:3: You already have grid-area ' +
+        'autoprefixer: <css input>:29:3: You already have a grid-area ' +
         'declaration present in the rule. You should use either ' +
         'grid-area or grid-column, not both'
       ])

--- a/test/cases/grid-area.css
+++ b/test/cases/grid-area.css
@@ -21,3 +21,10 @@
 .f {
   grid-area: 2 / 2 / 3;
 }
+
+/* emit warning */
+.g {
+  grid-area: 3 / 1;
+  grid-row: 1;
+  grid-column: 2;
+}

--- a/test/cases/grid-area.out.css
+++ b/test/cases/grid-area.out.css
@@ -38,3 +38,12 @@
   -ms-grid-column: 2;
   grid-area: 2 / 2 / 3;
 }
+
+/* emit warning */
+.g {
+  -ms-grid-row: 3;
+  -ms-grid-column: 1;
+  grid-area: 3 / 1;
+  grid-row: 1;
+  grid-column: 2;
+}

